rs/registry/admin/src/main.rs

```diff
@@ -5530,6 +5551,129 @@ async fn propose_to_add_or_remove_node_provider(
     };
 }
 
+async fn propose_to_install_code(
+    cmd: ProposeToChangeNnsCanisterCmd,
+    agent: Agent,
+    proposer: NeuronId,
+) {
+    let canister_client = GovernanceCanisterClient(NnsCanisterClient::new(
+        agent,
+        GOVERNANCE_CANISTER_ID,
+        Some(proposer),
+    ));
+
+    let canister_id = Some(PrincipalId::try_from(cmd.canister_id).unwrap());
+    let wasm_module = Some(
+        read_wasm_module(
+            &cmd.wasm_module_path,
+            &cmd.wasm_module_url,
+            &cmd.wasm_module_sha256,
+        )
+        .await,
+    );
+    let arg = cmd.arg.as_ref().map(|path| read_file_fully(path));
+    let skip_stopping_before_installing = Some(cmd.skip_stopping_before_installing);
+    let install_mode = match cmd.mode {
+        CanisterInstallMode::Install => Some(GovernanceInstallMode::Install as i32),
+        CanisterInstallMode::Reinstall => Some(GovernanceInstallMode::Reinstall as i32),
+        CanisterInstallMode::Upgrade => Some(GovernanceInstallMode::Upgrade as i32),
+    };
```

Comment:
```suggestion
    let install_mode = Some(GovernanceInstallMode::from(cmd.mode));
```

IMHO, people very rarely spin out too aggressively; it's usually the other way: too much in lining.

Reply:
I now moved the conversion from command to action into a separate method. Given that, I'm not sure moving the enum conversion is beneficial